src/CarbonAware.WebApi/src/Filters/HttpResponseExceptionFilter.cs

```diff
@@ -9,62 +9,75 @@ namespace CarbonAware.WebApi.Filters;
 public class HttpResponseExceptionFilter : IExceptionFilter
 {
     private ILogger<HttpResponseExceptionFilter> _logger;
-    private static readonly ActivitySource _activitySource = new ActivitySource(nameof(HttpResponseExceptionFilter));
```

Comment:
Was removing the activity needed for this or was it a 🚗 -by change?

Reply:
> Was removing the activity needed for this or was it a 🚗 -by change?

It was a need to capture the traceId based on the current activity.
There is going to be another story since all these calls (activity source) are useless in case of AppInsight integration. That integration doesn't use any of that (all the datasource calls are null, since there is no listener).
It is better to have a complete story that show why/where to put these calls since it works fine if you do integration with OpentTelemetry. But the way is done today here is noisy.

Reply:
There should be a single activity source for the entire CarbonAwareSDK library.

Azure Monitor is used to listen for OpenTelemetry data and send it to Application Insights:  https://docs.microsoft.com/en-us/azure/communication-services/quickstarts/telemetry-application-insights?pivots=programming-language-csharp

So the calls aren't useless for appinsights.  Direct app insight is the "old" way of doing things and has been superseded by OpenTelemetry.

Reply:
Also, the appropriate thing to do to make the tracing stuff not null would be to add a console exporter.  See here for examples:  https://opentelemetry.io/docs/instrumentation/net/getting-started/

Note that this should only be done in a development environment or in environments where console output isn't treated as logs (like k8s).